### PR TITLE
GH-12194 Converting to structured logging the file app/user.go

### PR DESCRIPTION
**Pull request by DarrellRichards**
Summary
Converting to structured logging the file app/user.go

Ticket Link
Fixes: #12194

### CODE REVIEW

File: app/user.go [hunk 1]
```diff
@@ -397,7 +397,7 @@ func (a *App) CreateOAuthUser(service string, userData io.Reader, teamId string)
 
 		err = a.AddDirectChannels(teamId, user)
 		if err != nil {
-			mlog.Error(err.Error())
+			mlog.Error("There was a error with adding direct channel", mlog.Err(err))
```
Comment (hanzei) [inline app/user.go]: For consistency I would like to go with 
```suggestion
			mlog.Error("Failed to add direct channels", mlog.Err(err))
```

File: app/user.go [hunk 2]
```diff
@@ -2242,29 +2242,29 @@ func (a *App) PromoteGuestToUser(user *model.User, requestorId string) *model.Ap
 	for _, team := range userTeams {
 		// Soft error if there is an issue joining the default channels
 		if err = a.JoinDefaultChannels(team.Id, user, false, requestorId); err != nil {
-			mlog.Error(fmt.Sprintf("Encountered an issue joining default channels err=%v", err), mlog.String("user_id", user.Id), mlog.String("team_id", team.Id), mlog.String("requestor_id", requestorId))
+			mlog.Error("Encountered an issue joining default channels", mlog.String("user_id", user.Id), mlog.String("team_id", team.Id), mlog.String("requestor_id", requestorId), mlog.Err(err))
```
Comment (hanzei) [inline app/user.go]:
```suggestion
			mlog.Error("Failed to joining default channels", mlog.String("user_id", user.Id), mlog.String("team_id", team.Id), mlog.String("requestor_id", requestorId), mlog.Err(err))
```

**Comment by DarrellRichards** [comment]
@hanzei Thanks for the feedback, I'll keep the wording suggestions in mind as I work on other PR's.

Thanks again.

**Comment by DSchalla** [comment]
Thank you very much @DarrellRichards! Adds a lot of context to the errors itself. :-)

**Pull request merged**